test/Uri.js

```diff
@@ -48,8 +48,19 @@ describe('Uri', function() {
 		assert.strictEqual('', uri.getOrigin());
 	});
 
+	it('should set procotol http: on localhost uri', function() {
+		var uri = new Uri('localhost:8080');
+		assert.strictEqual('http:', uri.getProtocol());
+		assert.strictEqual('8080', uri.getPort());
+	});
+
+	it('should support all types of uri schemes', function() {
+		var uri = new Uri('test:hostname');
+		assert.strictEqual('test:', uri.getProtocol());
+	});
+
 	it('should support set port on uri', function() {
-		var uri = new Uri('hostname:8080');
```

Comment:
@Robert-Frampton,

This method adds the default protocol in the case that the protocol isn't defined.  The issue with the way it was previously done is that it would add a protocol to URIs that already have a protocol. 

.i.e. `tel:123-123-1234`

The changes here loosen the restriction as to what a valid protocol is and checks against it.